aea/configurations/data_types.py

```diff
@@ -451,13 +455,10 @@ def __repr__(self) -> str:
         return f"<{self}>"
 
     def __eq__(self, other: Any) -> bool:
-        """Compare with another object."""
-        return (
-            isinstance(other, PublicId)
-            and self.author == other.author
-            and self.name == other.name
-            and self.version == other.version
-        )
+        """Check equality."""
+        if not isinstance(other, self.__class__):
+            return NotImplemented  # Delegate comparison to the other instance's __eq__.
+        return all(getattr(self, s) == getattr(other, s) for s in self.__slots__)
 
```

Comment:
https://github.com/valory-xyz/open-aea/blob/8e3ce14cd1c2bac357ceab45fbe7215769abf7df/aea/configurations/data_types.py#L244

Reply:
This makes cli command and related check and tests fail because of the additional hash being included. I am going to exclude it from the check, since hash of the component itself it not present in it's .yaml file